content/2016.12.how-to-develop-a-webapp-with-Kinto.md

```diff
@@ -0,0 +1,247 @@
+    "settings": {
+        "readonly": false,
+        "batch_max_requests": 25
+    },
+    "capabilities": {
+        "admin": {
+            "description": "Serves the admin console.",
+            "version": "1.5.0",
+            "url": "https://github.com/Kinto/kinto-admin/"
+        },
+        "default_bucket": {
+            "description": "The default bucket is an alias for a personal bucket where collections are created implicitly.",
+            "url": "https://kinto.readthedocs.io/en/latest/api/1.x/buckets.html#personal-bucket-default"
+        }
+    },
+    "project_docs": "https://kinto.readthedocs.io/",
+    "project_name": "kinto",
+    "http_api_version": "1.12",
+    "url": "http://kinto.discute.re/v1/"
+
+}
+```
+
+And that's it! Now we can move to the front-end development phase.
+
+## Viva JavaScript!
+
+In order to get the messages, we will have to write a bit of code to get the room's id through the application URL:
+
+```js
+// https://discute.re/#4815162342  is mapped to the collection 4815162342
+this.collectionName = window.location.hash.slice(1);
+```
+
+Then we need to do a GET request to get the list of all room's messages.
+A simple `GET https://kinto.discute.re/v1/buckets/chat/collections/4815162342/records` would do the job, but we will rather be using the [kinto-https-js](https://github.com/kinto/kinto-http.js) library that offers a clean API, take a look at this code ma’:
+
+    const kinto_url = 'https://kinto.discute.re/v1';
+    // options stores the connection identifiers
+    const client = new KintoClient(kinto_url, options);
+    client.bucket('chat')
+        .collection('4815162342')
+        .listRecords()
+        .then(({data}) => displayData(data));
+
+The `options` variable contains at least the authentication token. This token was not created before. Kinto accepts all requests and responds accordingly to the token: if the token gives access to the resource, the user can act on it (create, update, etc.), otherwise he can't.
+[Read more about authentication here](https://kinto.readthedocs.io/en/stable/api/1.x/authentication.html).
+
+```js
+displayData(data) can now retrieve the JSON object and render it as HTML.
+[
+    {
+        "message": "I know nothing",
+        "last_modified": 1480340246147,
+        "author": "Jon Snow",
+        "id": "a1e6e879-5c94-4a01-9ec2-28e152d1d8f4"
+    },
+    {
+        "message": "I know that I know nothing",
+        "last_modified": 1480340396521,
+        "author": "Socrates",
+        "id": "94787c48-aa96-4121-879f-1bd68d3c1f23"
+    }
+]
+
+getMessage(message, author) {
+    return (
+    `<li>
+      <span>
+        <span>${author}</span>
+        <span>${message}</span>
+      </span>
+    </li>`
+    );
+}
+
+displayMessage({message, author}) {
+    var container = document.createElement('div');
+    container.innerHTML = this.getMessage(message, author);
+    this.messagesList.firstChild.appendChild(container.firstChild);
```

Comment:
What context is `this`? Is it meant to reference `window`? This is very unclear :/

Why don't we just skip using it at all?